lib/images.js

```diff
@@ -24,62 +24,46 @@ const storage = gcloud.storage({
   projectId: config.get('GCLOUD_PROJECT')
 });
 const bucket = storage.bucket(CLOUD_BUCKET);
+const fetch = require('node-fetch');
 
-// Returns the public, anonymously accessable URL to a given Cloud Storage
-// object.
-// The object's ACL has to be set to public read.
-// [START public_url]
-function getPublicUrl(filename) {
-  return 'https://storage.googleapis.com/' + CLOUD_BUCKET + '/' + filename;
+function fetchAndSave(url, destFile) {
+  return new Promise((resolve, reject) => {
+    fetch(url)
+      .then(response => {
+        return saveImage(response, destFile);
+      })
+      .then(url => {
+        resolve(url);
+      })
+      .catch(err => {
+        reject.log(err);
+      });
+  });
 }
-// [END public_url]
 
-// Express middleware that will automatically pass uploads to Cloud Storage.
-// req.file is processed and will have two new properties:
-// * ``cloudStorageObject`` the object name in cloud storage.
-// * ``cloudStoragePublicUrl`` the public url to the object.
-// [START process]
-function sendUploadToGCS(req, res, next) {
-  if (!req.file) {
-    return next();
-  }
-
-  const gcsname = Date.now() + req.file.originalname;
-  const file = bucket.file(gcsname);
-  const stream = file.createWriteStream();
-
-  stream.on('error', err => {
-    req.file.cloudStorageError = err;
-    next(err);
-  });
+function saveImage(response, destFile) {
```

Comment:
@juliantoledo @andreban WDYT about adding jsdoc to this and other functions? It would be useful to know what response is (a `http.IncomingMessage`?) and what `destFile` is (a string?), what the return value is, etc.